lnwallet/channel.go

```diff
@@ -2139,10 +2139,19 @@ func (lc *LightningChannel) createCommitmentTx(c *commitment,
 
 	// Currently, within the protocol, the initiator always pays the fees.
 	// So we'll subtract the fee amount from the balance of the current
-	// initiator.
-	if lc.channelState.IsInitiator {
+	// initiator. If the initiator is unable to pay the fee fully, then
+	// their entire output is consumed.
+	switch {
+	case lc.channelState.IsInitiator && commitFee > ourBalance.ToSatoshis():
+		ourBalance = 0
```

Comment:
hmm, is this even a legal state? If the initiator doesn't have enough balance to cover the fee, then this commitment tx should never have been created

Reply:
I believe it is possible in certain cases when the fee rate is updated. See https://github.com/lightningnetwork/lightning-rfc/blob/master/02-peer-protocol.md#updating-fees-update_fee.

Reply:
It can happen at times with concurrent fee updates by the initiator and new state transitions by the responder. It's an edge case in the spec that isn't _suuuper_ well defined atm, but it can happen. 

After I updated to use `uint64` everywhere, the test for this case that @jimpo added failed until I added this condition.